lib/policyEvaluator/utils/conditions.js

```diff
@@ -0,0 +1,407 @@
+ * if necessary
+ * @param {string | array} time - value or values to be converted
+ * @return {string | array} converted value or values
+ */
+function convertToEpochTime(time) {
+    function convertSingle(item) {
+        // If ISO time
+        if (item.indexOf(':') > -1) {
+            return new Date(item).valueOf();
+        }
+        return item;
+    }
+    if (!Array.isArray(time)) {
+        return convertSingle(time);
+    }
+    return time.map(single => convertSingle(single));
+}
+
+/**
+ * convertConditionOperator converts a string operator into a function
+ * each function takes a string key and array of values as arguments.
+ * Variables in the value are handled before calling this function but
+ * wildcards and switching ${$}, ${*} and ${?} are handled here because
+ * whether wildcards allowed depends on operator
+ * @param {string} operator - condition operator
+ * Possible Condition Operators:
+ * (http://docs.aws.amazon.com/IAM/latest/UserGuide/
+ * reference_policies_elements.html)
+ * @return {boolean} true if condition passes and false if not
+ */
+conditions.convertConditionOperator = operator => {
+    // Policy Validator checks that the condition operator
+    // is only one of these strings so should not have undefined
+    // or security issue with object assignment
+    const operatorMap = {
+        StringEquals: function stringEquals(key, value) {
+            return value.some(item => {
+                const swtichedOutChars = convertSpecialChars(item);
+                return swtichedOutChars === key;
+            });
+        },
+        StringNotEquals: function stringNotEquals(key, value) {
+            // eslint-disable-next-line new-cap
+            return !operatorMap.StringEquals(key, value);
+        },
+        StringEqualsIgnoreCase: function stringEqualsIgnoreCase(key, value) {
+            const lowerKey = key.toLowerCase();
+            return value.some(item => {
+                const swtichedOutChars = convertSpecialChars(item);
+                return swtichedOutChars.toLowerCase() === lowerKey;
+            });
+        },
+        StringNotEqualsIgnoreCase:
+            function stringNotEqualsIgnoreCase(key, value) {
+                // eslint-disable-next-line new-cap
+                return !operatorMap.StringEqualsIgnoreCase(key, value);
+            },
+        StringLike: function stringLike(key, value) {
+            return value.some(item => {
+                const wildItem = handleWildcards(item);
+                const wildRegEx = new RegExp(wildItem);
+                return wildRegEx.test(key);
+            });
+        },
+        StringNotLike: function stringNotLike(key, value) {
+            // eslint-disable-next-line new-cap
+            return !operatorMap.StringLike(key, value);
+        },
+        NumericEquals: function numericEquals(key, value) {
+            return value.some(item => item === key);
```

Comment:
As other numeric functions are doing, you would have to convert the key and value to numbers if they are not. It's so that you can avoid comparisons of (string, number) or (number, string)